### DISCUSSION TITLE
Can you please provide a .net version of this Java-dsl library

### DISCUSSION THREAD

**Discussion by sooraj171** [abstracta/jmeter-java-dsl#181] [Ideas] [answered]
Can you please provide a .net version of this Java-dsl library; 
So we can add this library to .net project directly and it generate the Jmeter code after .net tests are completed?

**Comment by rabelenda** [comment]
Thank you very much @sooraj171 for asking for this! 

I think is a great idea. We have had thoughts about this, but have prioritized other features over this, "waiting" (in some sense) for the community to demonstrate interest in this before actually investing on it. So, if anyone else also thinks this would be good/interesting **please let us know by upvoting this discussion!**

The last part of your question "and it generate the Jmeter code after .net tests are completed" I guess is related to the Selenium Conf demo and [this project](https://github.com/abstracta/selenium-jmeter-dsl-demo).

If so, then what you require would require 3 steps:
1. Implement a .net JMeter DSL: we have thought about this, and have some ideas on how this could be achieved taking advantage of what we have already built with jmeter-java-dsl. This also nicely fit on our general vision of the project, to start with a solid java API that we and the rest of the community can use later on to create DSLs for other languages. Like @eldaduzman has done with [pymeter](https://github.com/eldaduzman/pymeter/).
2. As to be able to record and generate .net code, we would need to extend jmx2dsl to be able to generate .net code in addition to java code
3. Make some integration example as [this project](https://github.com/abstracta/selenium-jmeter-dsl-demo) with .net that gets you the full cycle that I think you are referring to.

Thank you again for asking about this! And please, if anyone is interested on this upvote this discussion to help us prioritize it higher!

**Reply by eldaduzman** [reply]
Hi @rabelenda !

If we wish to keep the same strategy as taken in Pymeter, we need to find a bridge between C# and java.
I'm not so good in C# but I remember looking at such bridges years ago.

**Reply by rabelenda** [reply]
Yes, we have done some initial research about it, and are considering implementing some PoCs to identify the best option. If someone wants to submit a PR for this we would also consider submitted options.

**Comment by sooraj171** [comment]
@rabelenda any update on this topic?

**Reply by rabelenda** [reply, marked as answer]
Hello, we just released an initial version of [.Net DSL](https://abstracta.github.io/jmeter-dotnet-dsl/). 

Feel free to ask for features and help improving it.

**Reply by rabelenda** [reply]
Any thoughts on this? :)

**Reply by sooraj171** [reply]
This is great!!. Will review more and will let you know.
